packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -206,14 +210,15 @@ export class RunningSummarizer implements IDisposable {
     }
 
     public handleOp(error: any, { sequenceNumber, type, clientId, contents }: ISequencedDocumentMessage) {
+        this.numNonSystemOps++;
         if (error !== undefined) {
             return;
         }
         this.heuristicData.lastOpSequenceNumber = sequenceNumber;
 
         // Check for enqueued on-demand summaries; Intentionally do nothing otherwise
         if (!this.tryRunEnqueuedSummary()) {
-            this.heuristicRunner?.run();
+            this.heuristicRunner?.run(this.numSystemOps, this.numNonSystemOps);
```

Comment:
I have a feeling that it's better to move all the logic around tracking ops (including op handler) into this.heuristicRunner.

Reply:
I don't know if I'm a huge fan of this idea. I see the heuristic runner as determining, based on the current data, whether a summary should be run or not. In my opinion, it shouldn't also be the one in charge of collecting/manipulating this data.
The only thing it should be "running" is very specific heuristic runners that will self-actuate (ex: idle timer).